code/modules/mob/living/simple_animal/hostile/spaceworms.dm

```diff
@@ -0,0 +1,298 @@
+
+	minbodytemp = 0
+	maxbodytemp = 350
+	min_oxy = 0
+	max_co2 = 0
+	max_tox = 0
+
+	a_intent = "harm" //so they don't get pushed around
+
+	environment_smash = 1
+
+	speed = -1
+
+	AIenabled = 0 //The body isn't conscious
+
+	faction = list("spaceworms")
+
+	var/mob/living/simple_animal/hostile/spaceWorm/previousWorm //next/previous segments, correspondingly
+	var/mob/living/simple_animal/hostile/spaceWorm/nextWorm     //head is the nextest segment
+
+	var/mob/living/simple_animal/hostile/spaceWorm/wormHead/myHead		//Can be the same as next, just a general reference to the main worm.
+
+	var/stomachProcessProbability = 50
+	var/digestionProbability = 20
+
+	var/atom/currentlyEating //what the worm is currently eating
+	var/plasmaPoopPotential = 5 //this mainly exists for the name
+
+
+//Worm Head, Controls the AI for the entire worm "entity"
+/mob/living/simple_animal/hostile/spaceWorm/wormHead
+	name = "space worm head"
+	icon_state = "spacewormhead"
+	icon_living = "spacewormhead"
+	icon_dead = "spacewormdead"
+
+	//Stronger than normal segments
+	maxHealth = 125
+	health = 125
+
+	melee_damage_lower = 20
+	melee_damage_upper = 25
+	attacktext = "bites"
+
+	animate_movement = SLIDE_STEPS
+
+	AIenabled = 1//The head is conscious
+	stop_automated_movement = 0 //Ditto ^
+
+	//head specific variables
+	var/spawnWithSegments = 6
+	var/list/totalWormSegments = list() //doesn't contain src
+	var/catastrophicDeathProb = 15 //15% chance for the death of the head to kill the whole thing
+
+
+/mob/living/simple_animal/hostile/spaceWorm/wormHead/New(var/location, var/segments = spawnWithSegments)
+	..()
+
+	if(!src)//This is to prevent a runtime.
+		return
+
+	myHead = src //It's it's own head.
+
+	//Used in the for to attach each worm segment to the next in the sequence, instead of all of them to src
+	var/mob/living/simple_animal/hostile/spaceWorm/currentWormSeg = src
+
+	for(var/i = 1 to segments)
+		var/mob/living/simple_animal/hostile/spaceWorm/newSegment = new /mob/living/simple_animal/hostile/spaceWorm(loc)
+		currentWormSeg.Attach(newSegment)
+		currentWormSeg = newSegment
+
+	for(var/mob/living/simple_animal/hostile/spaceWorm/SW in totalWormSegments)
+		SW.update_icon()
+
+/mob/living/simple_animal/hostile/spaceWorm/wormHead/update_icon()
+	if(stat == CONSCIOUS || stat == UNCONSCIOUS)
+		icon_state = "spacewormhead[previousWorm ? 1 : 0]"
+		if(previousWorm)
+			dir = get_dir(previousWorm,src)
+	else
+		icon_state = "spacewormheaddead"
+
+	for(var/mob/living/simple_animal/hostile/spaceWorm/SW in totalWormSegments)
+		if(SW == src)//incase src ends up in here we don't want an infinite loop
+			continue
+		SW.update_icon()
+
+
+//Try to move onto target's turf and eat them
+/mob/living/simple_animal/hostile/spaceWorm/wormHead/AttackingTarget()
+	..()
+	attemptToEat(target)
+
+//Attempt to eat things we bump into, Mobs, Walls, Clowns
+/mob/living/simple_animal/hostile/spaceWorm/wormHead/Bump(atom/obstacle)
+	attemptToEat(obstacle)
```

Comment:
That's why it's also in AttackingTarget()
The AI didn't bump often, but with a client they did so, it's like spider mob verbs, it's more for the players.